### PR TITLE
fix: optimize verification level skip check

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

**Review by rgnote — APPROVED** [review]
LGTM

File: notation.go [hunk 1]
```diff
@@ -194,17 +198,18 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, re
 	}
 
 	// passing nil signature to check 'skip'
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply notation.go:200]: Updated.

[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply notation.go:203]: Updated.

File: verifier/verifier.go [hunk 2]
```diff
@@ -62,6 +62,27 @@ func New(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore
...
+		return false, nil, notation.ErrorNoApplicableTrustPolicy{Msg: err.Error()}
+	}
+	logger.Debugf("Trust policy configuration: %+v", trustPolicy)
+	// ignore the error since we already validated the policy document
+	verificationLevel, _ := trustPolicy.SignatureVerification.GetVerificationLevel()
+
+	// verificationLevel is skip
+	if reflect.DeepEqual(verificationLevel, trustpolicy.LevelSkip) {
+		logger.Debug("Skipping signature verification")
+		return true, verificationLevel, nil
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply verifier/verifier.go]: Updated.

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**